crates/ruma-events/src/room/matrix_rtc.rs

```diff
@@ -49,6 +87,50 @@ pub struct Membership {
     pub membership_id: String,
 }
 
+impl Membership {
+    /// Application is "m.call" and scope is "m.room"
+    pub fn is_room_call(&self) -> bool {
+        if let Application::Call(call) = &self.application {
+            return call.scope == CallScope::Room;
+        }
+        false
+    }
+
+    /// Application is "m.call"
+    pub fn is_call(&self) -> bool {
+        match self.application {
+            Application::Call(_) => true,
+            _ => false,
+        }
+    }
+
+    /// Check if the event is expired.
+    /// Defaults to using `created_ts` in the event content.
+    /// If no `origin_server_ts` is provided and the event does not contain `created_ts`
+    /// the event will be considered as not expired. (A warning will be logged)
+    /// # Arguments
+    ///  - [`origin_server_ts`] a fallback if `created_ts` is not present
+    pub fn is_expired(&self, origin_server_ts: Option<MilliSecondsSinceUnixEpoch>) -> bool {
+        let ev_created_ts = match (self.created_ts, origin_server_ts) {
+            (Some(created_ts), Some(_)) => Some(created_ts),
+            (None, Some(server_ts)) => Some(server_ts),
+            (Some(created_ts), None) => Some(created_ts),
+            _ => None,
+        };
+        if let Some(ev_created_ts) = ev_created_ts {
+            let now = MilliSecondsSinceUnixEpoch::now().to_system_time();
+            let expire_ts =
+                ev_created_ts.to_system_time().map(|t| t + Duration::from_millis(self.expires));
+            now > expire_ts
+        } else {
+            // This should not be reached since we only allow events that have copied over
+            // the origin server ts. `copy_origin_server_ts_to_membership`
+            warn!("Encountered a Call Member state event where the origin_ts (or origin_server_ts) could not be found.
+            It is treated as a non expired event but this might be wrong.");
+            false
+        }
+    }
+}
 
 /// Description of the SFU/Foci a membership can be connected to
```

Comment:
"foci" as in the plural of "focus"?

Reply:
You reacted with thumbs-up, but I don't think that really is a sufficient reply 😅

I really have no clue what a "focus" is in this context. Is it explained in some doc comment? (also `Foci` should not be captialized in that case, right?)

Reply:
Foci is the plural of focus and it is the term we use for a single forwarding unit. (I think the idea is to not enforce the use of an SFU, theoretically it could also me a MCU (multipoint control unit)

In other words: I should add more doc here?

Reply:
Yes, please.